### PR TITLE
pass soa_dir into guess_cluster/get_default_cluster_for_service.

### LINKED ISSUE #702: local-run ignores -y for guessing cluster
In [guess_cluster](https://github.com/Yelp/paasta/blob/bff8fe687c59075236a46b6ea479bf98f4af8f59/paasta_tools/cli/utils.py#L328) and [get_default_cluster_for_service](https://github.com/Yelp/paasta/blob/1f4b49a51fdc0f49dce2b38c970e502231b3df4a/paasta_tools/utils.py#L1192) we aren't passing around soa_dir, which means that the cluster-guessing functionality ignores -y/--yelpsoa-root

### PR DESCRIPTION
**Pull request by EvanKrall**
fixes #702

### CODE REVIEW

File: paasta_tools/cli/utils.py [hunk 1]
```diff
@@ -303,7 +303,7 @@ def guess_service_name():
     return os.path.basename(os.getcwd())
 
 
-def guess_instance(service, cluster, args):
+def guess_instance(service, cluster, args, soa_dir=DEFAULT_SOA_DIR):
```
Comment (nhandler) [inline paasta_tools/cli/utils.py]: Just steal the `soa_dir` from `args.yelpsoa_config_root` like we do on line 314

File: paasta_tools/cli/utils.py [hunk 2]
```diff
@@ -325,13 +325,13 @@ def guess_instance(service, cluster, args):
     return instance
 
 
-def guess_cluster(service, args):
+def guess_cluster(service, args, soa_dir=DEFAULT_SOA_DIR):
```
Comment (nhandler) [inline paasta_tools/cli/utils.py]: As above, use `args.yelpsoa_config_root`

**Comment by nhandler** [comment]
Fix and Ship

File: paasta_tools/cli/utils.py (hunk 1 above)
Comment (EvanKrall) [inline paasta_tools/cli/utils.py]: oh oops I didn't mean to change this line

Comment (EvanKrall) [inline paasta_tools/cli/utils.py]: but I probably should go ahead and make guess_instance do the right thing too

**Pull request merged**